src/H5Z.c

```diff
@@ -621,13 +616,8 @@ H5Z__flush_file_cb(void *obj_ptr, hid_t H5_ATTR_UNUSED obj_id, void H5_ATTR_PARA
                 /* Set the "sanity checked" flag */
                 object->sanity_checked = TRUE;
             } /* end if */
-
-            /* Check whether to use the collective metadata read DXPL */
-            coll_md_read = H5F_COLL_MD_READ(f);
-            if (H5P_USER_TRUE == coll_md_read)
-                H5CX_set_coll_metadata_read(TRUE);
```

Comment:
This code used to see if the user had set the file-global flag for metadata reads, sinze H5Zunregister doesn't take an Access Property List. It would then set collective metadata reads on the context, so that the flush below takes them into account. However, with everything now using the new H5F_get_coll_metadata_reads() routine, the flushing below should be directly aware of if the user set the file-global flag.